### ISSUE TITLE
Parcel fails to build JSX components in node_modules

### ISSUE THREAD

**Issue by mlowijs** [parcel-bundler/parcel#7925]
# 🐛 bug report

I'm trying to build and run a React application that uses @glyphs/react-core for icons. However, when starting the application, Parcel gives a build error on a random component in this module. JSX components that I created in the project itself work just fine.

## 🎛 Configuration (.babelrc, package.json, cli command)
Configuration for minimal example:

package.json
```js
{
  "name": "parcel2test",
  "version": "1.0.0",
  "description": "",
  "source": "index.html",
  "scripts": {
    "start": "parcel -p 6000"
  },
  "author": "",
  "license": "ISC",
  "dependencies": {
    "@glyphs/react-core": "^0.0.8",
    "react": "^18.0.0",
    "react-dom": "^18.0.0"
  },
  "devDependencies": {
    "parcel": "^2.4.1"
  }
}
```

## 🤔 Expected Behavior

I'd expect the JSX components to be built just fine so I can use them in the app.

## 😯 Current Behavior

The error message:

```
> parcel -p 6000

Server running at http://localhost:6000
🚨 Build failed.

@parcel/transformer-js: Unexpected token '}'. Expected this, import, async, function, [ for array literal, { for object literal, @ for decorator, function, class, null, true, false,
number, bigint, string, regexp, ' for template literal, (, or an identifier

  /home/michiel/git/parcel2test/node_modules/@glyphs/react-core/src/icons/Square.jsx:10:198
     9 |         <>
  > 10 | necap} strokeLinejoin={strokeLinejoin} strokeWidth={strokeWidth || 3}/>
  >    |      ^
    11 |         </>
    12 |       )
```

## 💻 Code Sample

index.jsx
```js
import { createRoot } from "react-dom";
import { CoreArrow } from "@glyphs/react-core";

const root = createRoot(document.getElementById("root"));
root.render(<CoreArrow />);
```

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | 2.4.1 |
| Node             | 14.19.1 |
| npm/Yarn         | 8.5.5 |
| Operating System | Linux |

**Comment by mischnic** [comment]
JSX is not part of the JavaScript standard, so Parcel doesn't support using JSX inside node_modules. That package should transpile JSX

**Comment by mlowijs** [comment]
But Parcel does support using JSX outside node_modules? What is the difference functionally speaking?

**Comment by mischnic** [comment]
For node_modules, only preset-env is applied. Inside of your project, custom configs apply (decorators, custom Babel plugins, JSX, ...). This is because we want to discourage packages from publishing non-standard syntax on npm which would require you as a user to have some special config.

**Comment by TiagoCavalcante** [comment]
Astro allow this, Parcel should allow it too.

**Comment by LexSwed** [comment]
are there any solutions for when we are forced to use `npm` packages that are not transpiled?

**Comment by christopherbauer** [comment]
> JSX is not part of the JavaScript standard, so Parcel doesn't support using JSX inside node_modules. That package should transpile JSX

Hi Mischnic, this is a good posture to have regarding these practices, but is there any way for engineers using parcel to fix this besides finding a new library? Unfortunately this seems like a pretty consistent issue across many packages in the react-native space - including in the community packages - so any thoughts on how to circumvent the issue would be appreciated.

**Comment by DnEgorWeb** [comment]
So it looks like we're forced by Parcel to force those libraries to transpile, huh?

**Comment by AvnerCohen** [comment]
Another common use case is a local library you want to develop and consume inside another project. It's a really nice behavior to have a local setup that allows you to import JSX from within node_modules in such case.
I agree the default approach is good, but an override special-custom-way-out-of-it, would be nice..

**Comment by digitalmaxing** [comment]
Facing this issue as well
```
🚨 Build failed.

@parcel/transformer-js: Expression expected
node_modules/resume-pin/src/ResumePin.jsx:6:56
    5 |     <>
  > 6 |         <Notes sx={{ position: 'absolute', top: '13px' }} fontSize={fontSize} />
  >   |                                                        ^
    7 |         <PushPinOutlined fontSize={fontSize} />
    8 |     </>
```

  I am looking to either:
    - have a solution and use parcel
    - switch to vite or something else

**Comment by GeorchW** [comment]
Oh man... I'm encountering an increasing number of Parcel quirks that makes it hard to justify not switching to vite again.
I absolutely *love* parcel's zero-config approach, but working around these things requires me to introduce lots of config again.

In this case: For an internal dependency that's being used in multiple project and uses JSX, I now have to add a build step to the library. This also means that I can't just have an npm-git dependency, since I'd be missing the build step. So I'll have to add an internal package registry. This is all so much configuration that I need to do to be able to use a "zero-config" bundler. Oh, and if I want to have fast refresh when working on the library locally (when symlinked into node_modules of a webapp), I also need a separate watcher process that builds the library.

The alternative is to just switch to vite. Now, I've been there before, and I concluded that I don't like vite's network request waterfall, which can be much slower than parcel when dealing with many files. In theory, I also need a little more configuration, but in practice, I can remove quite a few workarounds that I needed to install to accommodate parcel.

>  This is because we want to discourage packages from publishing non-standard syntax on npm which would require you as a user to have some special config.

I don't think Parcel is in the situation of enforcing standards on NPM, when many other bundlers are supporting this. This is effectively just an annoyance to Parcel users.

It's "standards over usability", IMO contrary to Parcel's value proposition.
